Logistic guidelines for KWoC added in events' doc (#39)
kossiitkgp/docs: events/kwoc.md

## First things first 

- KWoC is a month long event, generally conducted in December.
- Registrations start atleast a week before the coding period(More than a week is preferred)
- Coding period is generally 4 weeks long
- Mid evals are held after around 2 weeks of coding period
- End evals are done after the coding period

## Pre-registration

- The following things should be ready and updated:-

  - Website with registration feature tested internally.
  - FAQs on the website
  - Student and Mentor Docs of KWoC. Make sure that the content of the doc is updated with the current rules of KWoC(especially rules like Mid Evals and End Evals criteria). If there are any new important changes, do mention them in the docs. 

> - A dedicated slack workspace for KWoC. If the old one has reached the maximum free limit, create a new Slack workspace.

Author: @Shankusu993 was suggesting about having a discord server. Advantages that he mentioned:
- No limit on number of users
- Easy to manage roles

But I say Slack is better because, for every query there can be a thread created. But discord doesn't have threads. After 10 or 20 messages, the query might be lost. And if a query is unanswered, the defeats the purpose of a platform for queries(other than mail/Github). Others, please give your opinion

Reviewer: Another important reason for me suggesting discord besides the no member limit is to avoid what happened a year ago where a person kept spamming the channel and created disturbance for everyone. When banned he created new accounts and proceeded to do the same. In discord it's easy to atleast  avoid spam part by having slow mode on a channel.

Author: > It doesn't make much of a difference

No, it makes a difference. Discord doesn't provide the feature of threads. If after a query there are some 50 messages on some other topic it gets lost. With threads that does't happen.

Also, I and @sahil-shubham checked. There is no limit on the number of users in slack. 

Spamming happens very rarely. It happened once in 2 years. That too on one part of the day. Clarifying doubts with ease is the main aim of a common group. We can't sacrifice it.

Reviewer 2: > No, it makes a difference.

Oh, yeah! It matters, my bad. I meant in a sense that it doesn't matter much to me or in the grand scheme of KWoC (except threads as you pointed out). So, I didn't care much.

- Any important changes or announcements(like informing in case of spam, changing Mid evals dates) are to be made on both website and via email.

## Mid Evals

- It is important to inform about the Mid evals and its criteria via email and KWoC website atleast 3 days before the Mid evaluation. 
- **Mid evals criteria**:
  - **KWoC 20 criteria**: Having atleast one commit before was the criteria for clearing Mid evals. If the mentor has been inactive and the PR remains unmerged. We asked the students to mail 
  - In the previous KWoCs, there was a form, which was to filled by mentors. However, to automate stuff the single commit criteria was implemented. 
  - You are encouraged to make changes in the Mid Eval criteria, but make sure that it is fair and is not too hard to clear. (Remember, KWoC is meant to encourage beginners)
- In case of serious emergencies, do not hesitate cancel mid evaluations. For example, in 2019 when there was a ban of internet in few states, the mid evals were cancelled. 
- In case, if any individual is unable to clear mid evals due to some genuine reasons, it should be considered.

## End evals

- **End evals criteria**:

>   - Till KWoC 20, the end evals criteria was asking the students to write a blog that summarizes their work. And the students would be evaluated based on the blog.

Reviewer 2: ```suggestion
  - Till KWoC 20 the end evals passing criteria was evaluating the students on a self-written blog summarising their work.
```

>   - The above criteria was made to make sure that no mentor would unfairly pass/fail a student as per his/her wish.

Reviewer 2: ```suggestion
  - The above criterion was made so that no mentor could unfairly pass/fail a student.
```

>   - Feel free to come up with any other criteria for End evals, but make sure that no student should have an unfair advantage.

Reviewer 2: ```suggestion
  - Feel free to come up with any other criteria for End evals given no student gets an unfair advantage.
```

> - If blog criteria is being followed

Reviewer 2: ```suggestion
- If blog criterion is being followed:
```

>   - Make sure to inform the students atleast a week time after the coding period ends to write the blog

Reviewer 2: ```suggestion
  - Inform the students within a week of the end of the coding period to write the blog.
```

>   - Inform the criteria for end evals via email. Examples of previous blogs should be given.

Reviewer 2: ```suggestion
  - Inform the criteria for end evals via email sharing previous passing blogs.
```

> - If  a student was unable to submit his/her blog before the deadline due to some genuine reasons. Do consider it later, its a month long effort!

Reviewer 2: ```suggestion
- Consider a blog submitted after the deadline if the student has a genuine reason for the delay.
```

- **Evaluation of blogs**:

>   - For evaluation, distribute the blogs among members of KOSS for evaluation

Reviewer 2: ```suggestion
  - Distribute the blogs among members of KOSS.
```

>   - While evaluating, if a member finds some blog that doesn't seem to clear the end evals, make sure to cross verify the same with other members of KOSS

Reviewer 2: ```suggestion
  - If a blog seems to not clear the end evals passing criteria, cross verify with other members.
```

> - It is good to inform the participants by what date, they might expect the results of KWoC after they have submitted their End evals Blog. 

Reviewer 2: ```suggestion
- Preferably inform the participants KWoC results date after they have submitted their blogs. 
```

## Certificates

> - Along with the name of the participant, the certificate must contain the following links

Reviewer 2: ```suggestion
- Along with the name of the participant the certificate must contain the following links:
```

>   - Verification link: A link of certificate hosted on a Public repository of KOSS - [kossiitkgp/public-files](https://github.com/kossiitkgp/public-files)

Reviewer 2: ```suggestion
  - Verification link: A link of the certificate hosted on a Public repository of KOSS - [kossiitkgp/public-files](https://github.com/kossiitkgp/public-files)
```

>   - Stats link: A link containing Student's stats in case of a student and Project(s) Stats in case of Mentor.

Reviewer 2: ```suggestion
  - Stats link: A link containing the Student's or the Project's stats respectively for a student or a Mentor.
```

> - Move the website from `kwoc.kossiitkgp.org` to `kwoc{Y}.kossiitkgp.org`, where {Y} takes the value of year conducted. This must be done so that the stats are hosted on latter domain and the former - `kwoc.kossiitkgp.org`  is reused next year.

Reviewer 2: ```suggestion
- Move the website from `kwoc.kossiitkgp.org` to `kwoc{Y}.kossiitkgp.org` (e.g. `kwoc19.kossiitkgp.org` for KWoC19). Thereafter, the stats are hosted on the latter while the former is reused next year.
```

Author: This is nice!

> - The certificates should be hosted on public files repo - [kossiitkgp/public-files](https://github.com/kossiitkgp/public-files)

Reviewer 2: ```suggestion
- The certificates must be hosted on the public files repo - [kossiitkgp/public-files](https://github.com/kossiitkgp/public-files)
```

- The certificates in PDF formats should be sent via mail to the participants.

## Miscellaneous

> - Take an anonymous feedback about the proceedings of KWoC from participants(preferably after end evals blog has been submitted).

Reviewer 2: ```suggestion
- Take anonymous feedback about the proceedings of KWoC from participants. Preferably after the submission of the end evals blog.
```

> - Update [kossiitkgp/mistakes](https://github.com/kossiitkgp/mistakes) repo for sake of posterity based on feedback and other things that went wrong.

Reviewer 2: ```suggestion
- Update [kossiitkgp/mistakes](https://github.com/kossiitkgp/mistakes) repo for the sake of posterity based on feedback and other things that went wrong.
```
